k2/csrc/fsa_algo.h

```diff
@@ -178,6 +178,10 @@ void AddEpsilonSelfLoops(FsaOrVec &src, FsaOrVec *dest,
                          of states are active.  The hash size used per FSA is 4
                          times (this rounded up to a power of 2), so this
                          affects memory consumption.
+       @param [in] allow_partial If true, we will treat all the states on the
+                         last frame to be final state. If false, we only
+                         care about the real final state in the decoding
+                         graph on the last frame when generating lattice.
          @param[out] out Output vector of composed, pruned FSAs, with same
```

Comment:
the way allow_patrial works in Kaldi is, it is conditional: we only make the all final if there was no final state active.

Reply:
The code is implemented as "only make the all final if there was no final state active".
It's controlled by variable "has_valid_final_arc".
I will change this comment.